controllers/common.go

```diff
@@ -19,50 +19,31 @@ import (
 
 var hashOptions = hashstructure.HashOptions{ZeroNil: true}
 
-// isSameApiDefinition calculates the hash of two ApiDefinitionSpec structs and returns true
-// if hashes are equal. It is used to prevent calling extra Update requests to Tyk Gateway
-// if there is no changes on resources.
-func isSameApiDefinition(apiDef1, apiDef2 *model.APIDefinitionSpec) bool {
-	api1Hash, err1 := hashstructure.Hash(apiDef1, hashstructure.FormatV2, &hashOptions)
+// calculateHashes calculates hashes of given two interfaces. Returns empty string for hash
```

Comment:
This function could be refactored to a single function that takes in only 1 argument and returns it's hash in the required format. Then it could be called twice where needed.

Reply:
Thanks! Implemented in [0365448](https://github.com/TykTechnologies/tyk-operator/pull/625/commits/0365448d7f6ea2d6ea9cf16a4aa4bc1a16a72ec7)